### PR TITLE
revert: boot provider state management

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Revert the state management changes on boot provider.

Commit that we're trying to revert: https://github.com/dailydotdev/apps/commit/80281fe95eafaefa87846dae7bff3f7e12037291#diff-0b8070cbc8731402622dbf6b309d2bcc2d5cc057982bdd85c8cb1925256bb058L135

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://revert-boot.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/contexts/AuthContext.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ import { AccessToken, Boot, Visit } from '../lib/boot';
 import { isCompanionActivated } from '../lib/element';
```
Comment (rebelchris) [inline packages/shared/src/contexts/AuthContext.tsx:17]: Generic thread:
Should we not also fix the implementation changes that used `!isAuthReady`

**Review by rebelchris — APPROVED** [review]
Non blocking question more?

File: packages/shared/src/contexts/AuthContext.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/contexts/AuthContext.tsx:17]: That has been reverted already in the fix for hydration.

**Review by nensidosari — APPROVED** [review]
Looks ok

**Review by capJavert — CHANGES_REQUESTED** [review]
On preview I get empty screen on homepage

**Comment by sshanzel** [comment]
> On preview I get empty screen on homepage

Yes, haven't done e2e yet as I just finished my meetings. Once the test cases have turned green for all of them, I will start the e2e.

**Comment by sshanzel** [comment]
@capJavert should be back by now.

File: packages/shared/src/contexts/BootProvider.tsx [hunk 2]
```diff
@@ -202,50 +199,51 @@ export const BootDataProvider = ({
...
   gqlClient.setHeader(
     'content-language',
     (user as Partial<LoggedUser>)?.language || ContentLanguage.English,
   );
 
+  useEffect(() => {
+    if (remoteData) {
+      setInitialLoad(initialLoad === null);
+      updateBootData(remoteData);
+    }
```
Comment (capJavert) [inline packages/shared/src/contexts/BootProvider.tsx:243]: This missing? https://github.com/dailydotdev/apps/commit/80281fe95eafaefa87846dae7bff3f7e12037291#diff-2279addce4c236245d34c6431e5269d47f22fd68c58f83c257dafa1b4e1b4599L197

File: packages/shared/src/contexts/AuthContext.tsx [hunk 3]
```diff
@@ -101,9 +101,9 @@ const logout = async (reason: string): Promise<void> => {
 export type AuthContextProviderProps = {
   user?: LoggedUser | AnonymousUser;
   isFetched?: boolean;
-  isPastRegistration?: boolean;
```
Comment (capJavert) [inline packages/shared/src/contexts/AuthContext.tsx:104]: We also have `setIsRegistration` in `AuthOptions.onProfileSuccess`? https://github.com/dailydotdev/apps/commit/80281fe95eafaefa87846dae7bff3f7e12037291#diff-0d72ac8004175d51542cd27ed8342d244a8e8e7f26876ad22bca3cb57a62ec9fR247

Comment (sshanzel) [reply packages/shared/src/contexts/AuthContext.tsx:104]: This is different; we use this to check whether the user should be logged out.

File: packages/shared/src/contexts/BootProvider.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/contexts/BootProvider.tsx:243]: We don't need that anymore. That's only for anonymous users changing their settings. They don't have access to that anymore.

**Review by capJavert — APPROVED** [review]
Did basic sanity check, not 100% if everything is reverted, but think you see it better anyway.

File: packages/shared/src/contexts/BootProvider.tsx [hunk 4]
```diff
@@ -269,13 +267,13 @@ export const BootDataProvider = ({
         getRedirectUri={getRedirectUri}
         loadingUser={!dataUpdatedAt || !user}
         loadedUserFromCache={loadedFromCache}
-        visit={bootData?.visit}
+        visit={remoteData?.visit}
```
Comment (sshanzel) [inline packages/shared/src/contexts/BootProvider.tsx:270]: If this actually works, my guess where the fix could be is the usage of remote data over the cached one stored from the state. Let's see.

**Pull request merged**